### PR TITLE
Add missing tests

### PR DESCRIPTION
**Pull request by idanovo**

### CODE REVIEW

**Review by nopcoder — APPROVED** [review]
LGTM

File: pkg/api/controller_test.go [hunk 1]
```diff
@@ -480,6 +480,24 @@ func TestController_LogCommitsPredefinedData(t *testing.T) {
...
+			expectedCommits: []string{"commit10"},
+			expectedMore:    false,
+			stopAt:          "main",
+		},
+		{
+			name:            "stop_at_branch_ref_expression",
+			expectedCommits: []string{"commit10", "commit9", "commit8"},
+			expectedMore:    false,
+			stopAt:          "main~2",
+		},
```
Comment (arielshaqed) [inline pkg/api/controller_test.go:500]: Honestly it's cool, but I see no reason to test each of these.  Any one of these tests proves that we Deref the ref.  A unit test of logging does not need to do anything beyond testing that this happens.  (All of these -- and more -- are tested in `parse_ref_test.go` and `resolve_ref_test.go`!)

**Review by arielshaqed — APPROVED** [review]
Neat, thanks!

File: pkg/api/controller_test.go (hunk 1 above)
Comment (idanovo) [reply pkg/api/controller_test.go:500]: But it doesn't test whether we dereference the StopAt flag.
If we were testing it, I wouldn't be able to merge my [PR](https://github.com/treeverse/lakeFS/pull/7222)

**Pull request merged**